crypto/bio/bio_test.cc

```diff
@@ -1095,3 +1095,133 @@ TEST(BIOTest, ReadWriteEx) {
   EXPECT_FALSE(BIO_read_ex(bio.get(), nullptr, 0, &read));
   EXPECT_EQ(read, (size_t)0);
 }
+
+TEST(BIOTest, TestPutsAsWrite) {
+  // By default, |BIO_puts| acts as |BIO_write|.
+  bssl::UniquePtr<BIO> bio(BIO_new(BIO_s_mem()));
+  ASSERT_TRUE(bio);
+
+  // Test basic puts and read  
+  uint8_t buf[32];
+  EXPECT_EQ(12, BIO_puts(bio.get(), "hello world\n"));
+  EXPECT_EQ(12, BIO_read(bio.get(), buf, sizeof(buf)));
+}
+
+TEST(BIOTest, TestPutsGetsCtrlCallbacks) {
```

Comment:
Since this ended up getting length with the different assertions I would split this into 3 different test functions covering puts, gets, and ctrl.